hal/mbed_sleep_manager.c

```diff
@@ -61,14 +61,18 @@ static sleep_statistic_t* sleep_tracker_add(const char* const filename)
         }
     }
 
-    debug("No free indexes left to use in mbed sleep tracker.\r\n");
+    MBED_ASSERT(false && "No free indexes left to use in mbed sleep tracker");
```

Comment:
Do we need an `MBED_ASSERT_MESSAGE`? The current style is to leave comments surrounding the assert.

fiy @0xc0170

Reply:
That would be nice, we dont have that yet?